airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/cursor_based/MySqlCursorBasedStateManager.java

```diff
@@ -0,0 +1,101 @@
+package io.airbyte.integrations.source.mysql.cursor_based;
+
+import com.google.common.collect.Lists;
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.integrations.source.mysql.internal.models.CursorBasedStatus;
+import io.airbyte.integrations.source.mysql.internal.models.InternalModels.StateType;
+import io.airbyte.integrations.source.relationaldb.CursorInfo;
+import io.airbyte.integrations.source.relationaldb.models.DbState;
+import io.airbyte.integrations.source.relationaldb.state.StreamStateManager;
+import io.airbyte.protocol.models.v0.AirbyteStateMessage;
+import io.airbyte.protocol.models.v0.AirbyteStateMessage.AirbyteStateType;
+import io.airbyte.protocol.models.v0.AirbyteStreamNameNamespacePair;
+import io.airbyte.protocol.models.v0.AirbyteStreamState;
+import io.airbyte.protocol.models.v0.ConfiguredAirbyteCatalog;
+import io.airbyte.protocol.models.v0.StreamDescriptor;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class MySqlCursorBasedStateManager extends StreamStateManager {
```

Comment:
This is similar to the `PostgresCursorBasedStateManager` class correct? Basically just to enable writing the `state_type` and `version` keys? Can we add a comment for that

IIRC we had a ticket to refactor this in `StreamStateManager` when we implemented source-mysql - do you think we should do that here?